### PR TITLE
gameManagment: Add basic skeleton for player, client and global datastucture

### PR DESCRIPTION
**Pull request by abhishek20123g**
These commits make the following changes
  * adds various file and packages which will be required to manage and maintain
    `gameManagment` Module.
  * adds a basic skeleton to a player, client and global datastructure. This basic
    skeleton is highly volatile in nature can be updated in future.

### CODE REVIEW

File: gameManagment/gameDataStructure/globalRoomDataStructure.py [hunk 1]
```diff
@@ -1,4 +1,71 @@
...
+        """
+        self.allConnectedPlayers = []
+        self.cardRemainingToDistribute = generateRandom52cards()
+        self.cardThrowingStartAt = stateConstant.NO_FIXED_STARTING_PLAYER
+        self.currentGameState = stateConstant.STATE_SET_TO_CARD_DISTRIBUTION
+        self.highestBid = 0
+        self.isBidOver = False
+        self.mainPlayer = stateConstant.MAIN_PLAYER_IS_NOT_ASSIGNED
+        self.maximumCardForGivenRound = 0
+        self.playerAvailableForBid = [0, 1, 2, 3, 4, 5]
```
Comment (keshu6598) [inline gameManagment/gameDataStructure/globalRoomDataStructure.py:24]: what is Its Role

Comment (keshu6598) [reply gameManagment/gameDataStructure/globalRoomDataStructure.py:24]: playerAvailableForBid not required

Comment (abhishek20123g) [reply gameManagment/gameDataStructure/globalRoomDataStructure.py:24]: This variable was there to determine which which player can bid and which player can not

File: gameManagment/gameDataStructure/globalRoomDataStructure.py [hunk 2]
```diff
@@ -1,4 +1,71 @@
...
+        self.roundWiningPlayerID = stateConstant.NO_FIXED_STARTING_PLAYER
+        self.suitWeight = [0, 0, 0, 0]
+        self.teamDistribution = [stateConstant.NO_TEAM_ASSIGNED] * 6
+        self.trumpSet = cardsConstant.NOT_ASSIGNED
+
+    def joinNewPlayer(self, player_info):
```
Comment (keshu6598) [inline gameManagment/gameDataStructure/globalRoomDataStructure.py]: player_name

File: gameManagment/gameDataStructure/globalRoomDataStructure.py [hunk 3]
```diff
@@ -1,4 +1,71 @@
...
+        self.cardThrowingStartAt = stateConstant.NO_FIXED_STARTING_PLAYER
+        self.currentGameState = stateConstant.STATE_SET_TO_CARD_DISTRIBUTION
+        self.highestBid = 0
+        self.isBidOver = False
+        self.mainPlayer = stateConstant.MAIN_PLAYER_IS_NOT_ASSIGNED
+        self.maximumCardForGivenRound = 0
+        self.playerAvailableForBid = [0, 1, 2, 3, 4, 5]
+        self.roundWiningPlayerID = stateConstant.NO_FIXED_STARTING_PLAYER
+        self.suitWeight = [0, 0, 0, 0]
+        self.teamDistribution = [stateConstant.NO_TEAM_ASSIGNED] * 6
```
Comment (keshu6598) [inline gameManagment/gameDataStructure/globalRoomDataStructure.py:27]: deep copy

File: gameManagment/gameDataStructure/playersDataStructure.py [hunk 4]
```diff
@@ -1,4 +1,34 @@
...
+        """
+        Provide card distribution to given player
+        """
+        pass
+
+    def canPlayerBid(self, new_bid):
+        """
+        Determine can player have new bid.
+        """
+        pass
```
Comment (keshu6598) [inline gameManagment/gameDataStructure/playersDataStructure.py:34]: add remove_card()

File: gameManagment/gameDataStructure/globalRoomDataStructure.py (hunk 3 above)
Comment (abhishek20123g) [reply gameManagment/gameDataStructure/globalRoomDataStructure.py:27]:
```
In [1]: a = [1] * 10                                                                                                                                                                                        

In [2]: a                                                                                                                                                                                                   
Out[2]: [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]

In [3]: a[3] = 2                                                                                                                                                                                            

In [4]: a                                                                                                                                                                                                   
Out[4]: [1, 1, 1, 2, 1, 1, 1, 1, 1, 1]

In [5]:  

```

**Pull request merged**